apps/platform-app/src/app/[locale]/(auth)/_actions/mutations/registerNewUser.ts

```diff
@@ -16,23 +23,35 @@ export const registerNewUser = withSentry(
       }
     }
 
+    if (linkedinUsername) {
+      userData.linkedinDetails = {
+        create: {
+          username: linkedinUsername,
+        },
+      }
+    }
+
     const createdUser = await prisma.user.create({
       data: userData,
       include: {
         githubDetails: true,
+        linkedinDetails: true,
       },
     })
     if (!createdUser) {
       throw new Error('Failed to create user')
     }
-    const devName = name ?? githubUsername ?? createdUser.email
+    const devName =
+      name ?? githubUsername ?? linkedinUsername ?? createdUser.email
+
+    const isSpecialist = provider === 'github' || provider === 'linkedin'
```

Comment:
if the role management would be moved to signin callback route, we would not have access to roles at this point. 
`registerNewUser.ts` runs before signin callback route

Reply:
can't we pass `role` to that registerNewUser method?

```
 async ({
    id: _id,
    provider,
    githubUsername,
    linkedinUsername,
    name,
    role,
    ...data
  }) => {
```

Reply:
https://github.com/nerdbord/good-dev-hunting-app/pull/353#discussion_r1839678026

I see another problem now.

**_If role management during sign-in/sign-up is moved to the callback route,_** 

- we lose the ability to determine the user type within the registerNewUser method. The function flow currently runs as follows: signin(provider, callbackRoute) -> signIn flow -> registerNewUser() -> callback route.

We can’t pass the role to registerNewUser because it hasn’t been specified yet; the user type (whether 'specialist' or 'hunter') can’t be identified until the callback route is loaded.

Reply:
But can't we launch sign-in with added `role` param? I mean, this way we could differ signing up as hunter, and as specialists. Especially if they will register using the same provider. Something like, `/sign-up?role=HUNTER`